### DISCUSSION TITLE
Functional Interface with Weight Input

### DISCUSSION THREAD

**Discussion by Wuyxin** [pyg-team/pytorch_geometric#2888] [Q&A]
Now I need to assign certain weights, say A, to a GNN Conv, which is computed by A = B + C where B and C are matrices.
And I want to get the gradient of the Conv's output with respect to the weight matrix B.  My current implementation is 
```
state_dict = Conv.state_dict()
state_dict[_dict] = A + B
Conv.load_state_dict(state_dict)
 ...
grad(loss, B, create_graph=True)
```
But it did not work as the backpropagation fails when loading the state dict. 
While PyTorch has implemented functional interface with weight input which can enable the backpropagation, i.e.,
```
output = F.conv2d(input, weight)
```
Did PyTorch Geometric have provided such functions so that I can realize the requirement above.

Note it is similar to the question https://discuss.pytorch.org/t/assign-variable-to-model-parameter/6150

**Comment by rusty1s** [comment]
We don't have a functional interface for this, I'm sorry. However, I still feel that this can be easily achieved via standard modules:
```python
conv = GCNConv(...)
conv.weight = A + B
conv(x, edge_index)
...
grad(loss, B, ...)
```
Let me know if this works for you!
